R/space_filling.R

```diff
@@ -41,6 +49,191 @@
+    )
+  }
+  UseMethod("grid_space_filling")
+}
+
+#' @export
+#' @rdname grid_space_filling
+grid_space_filling.parameters <- function(x,
+                                          ...,
+                                          size = 5,
+                                          type = "any",
+                                          variogram_range = 0.5,
+                                          iter = 1000,
+                                          original = TRUE) {
+  # test for NA and finalized
+  # test for empty ...
+  params <- x$object
+  names(params) <- x$id
+  grd <- make_sfd(
+    !!!params,
+    size = size,
+    type = type,
+    variogram_range = variogram_range,
+    iter = iter,
+    original = original
+  )
+  names(grd) <- x$id
+  grd
+}
+
+#' @export
+#' @rdname grid_space_filling
+grid_space_filling.list <- function(x, ..., size = 5, type = "any",
+                                    variogram_range = 0.5,
+                                    iter = 1000,
+                                    original = TRUE) {
+  y <- parameters(x)
+  params <- y$object
+  names(params) <- y$id
+  grd <- make_sfd(
+    !!!params,
+    size = size,
+    type = type,
+    variogram_range = variogram_range,
+    iter = iter,
+    original = original
+  )
+  names(grd) <- y$id
+  grd
+}
+
+
+#' @export
+#' @rdname grid_space_filling
+grid_space_filling.param <- function(x, ..., size = 5,
+                                     variogram_range = 0.5,
+                                     iter = 1000,
+                                     type = "any", original = TRUE) {
+  y <- parameters(list(x, ...))
+  params <- y$object
+  names(params) <- y$id
+  grd <- make_sfd(
+    !!!params,
+    size = size,
+    type = type,
+    variogram_range = variogram_range,
+    iter = iter,
+    original = original
+  )
+  names(grd) <- y$id
+  grd
+}
+
+sfd_types <-
+  c("any", "audze_eglais", "max_min_l1", "max_min_l2", "uniform",
+    "latin_hypercube", "max_entropy")
+
+make_sfd <- function(...,
+                     size = 5,
+                     type = "any",
+                     variogram_range = 0.5,
+                     iter = 1000,
+                     original = TRUE,
+                     call = caller_env()) {
+  type <- rlang::arg_match(type, sfd_types)
+  validate_params(..., call = call)
+  param_quos <- quos(...)
+  params <- map(param_quos, eval_tidy)
+  p <- length(params)
+
+  if (type %in% c("any", "audze_eglais", "max_min_l1", "max_min_l2", "uniform")) {
+    has_premade_design <- sfd::sfd_available(p, size, type)
+
+    if (has_premade_design) {
+      grid <- sfd::get_design(p, num_points = size, type = type)
+      # TODO value_seq should keep factor levels
+      vals <- purrr::map(params, ~ value_seq(.x, size))
+      vals <- purrr::map(vals, ~ base_recycle(.x, size))
+      grid <- sfd::update_values(grid, vals)
+      names(grid) <- names(params)
+    } else {
```

Comment:
@hfrick Here's where it falls back to a ME design if there is no pre-made design. I can update the wording in the man file to be more specific.